### PR TITLE
Add keyboard shortcuts, switch to giu fork, clean up UI code

### PR DESCRIPTION
**Pull request by ianling**
- I forked and refactored giu, and switched HellSpawner to use that fork. (I also added window focus tracking to giu so that HellSpawner can determine which window has focus)
- I forked one of giu's dependencies, imgui-go, and merged some changes that AllenDang made in the giu repo into the imgui-go fork.
- Refactors and simplifies some UI stuff so making things like editor windows is simpler.
- Adds support for global and window-specific keyboard shortcuts. I added some global shortcuts to some menubar items, as well as some example shortcuts to the DT1 viewer.
- A couple bug fixes and fixes for race conditions here and there, whatever I ran into while testing these changes.

Stuff you should look at if you're QA'ing this code:

Test the keyboard shortcuts:

- Ctrl+Shift+N -- new project
- Ctrl+O -- open project
- Alt+P -- preferences modal
- Alt+Q -- quit (we can get rid of this one if you want, it was mostly just for testing)
- F1 -- about modal
- Escape -- closes any modals that are open (preferences, about, or properties)
- Ctrl+Shift+M -- Toggles MPQ viewer
- Ctrl+Shift+P -- Toggles Project explorer
- Ctrl+W -- closes the focused editor window
- Left arrow -- in the DT1 editor, goes to the previous tile group
- Right arrow -- in the DT1 editor, goes to the next tile group

Verify that the global shortcuts (the ones that are not specific to a particular editor window -- so, all of them but the last two) function appropriately at all times. Verify that the window-specific shortcuts only work when that window has focus.

Verify all the viewers still work correctly. I went through all of them myself and didn't see any issues, but it's good to get two sets of eyes on stuff

### CODE REVIEW

File: hswidget/cofviewer.go [hunk 1]
```diff
@@ -4,9 +4,9 @@ import (
 	"fmt"
 	"strconv"
 
-	"github.com/AllenDang/giu"
 	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2enum"
 	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2fileformats/d2cof"
+	"github.com/ianling/giu"
```
Comment (gravestench) [inline hswidget/cofviewer.go:9]: not really necessary, but no biggie

File: hswindow/hsdialog/hsprojectpropertiesdialog/projectpropertiesdialog.go [hunk 2]
```diff
@@ -73,14 +72,10 @@ func (p *ProjectPropertiesDialog) Show(project *hsproject.Project, config *hscon
...
-	}
-
+func (p *ProjectPropertiesDialog) Build() {
 	canSave := len(strings.TrimSpace(p.project.ProjectName)) > 0
 
-	hswidget.ModalDialog("Select Auxiliary MPQ##ProjectPropertiesSelectAuxMPQDialog", &p.mpqSelectDialogVisible, g.Layout{
+	p.IsOpen(&p.mpqSelectDialogVisible).Layout(g.Layout{
 		g.Child("ProjectPropertiesSelectAuxMPQDialogLayout").Size(300, 200).Layout(g.Layout{
 			g.ListBox("ProjectPropertiesSelectAuxMPQDialogItems", p.auxMPQNames).Border(false).OnChange(func(selectedIndex int) {
 				p.mpqSelectDlgIndex = selectedIndex
```
Comment (gravestench) [inline hswindow/hsdialog/hsprojectpropertiesdialog/projectpropertiesdialog.go:81]: we probably should rename `giu.PopupModal.IsOpen` to something else. I _think_ that it's using the input argument as the control value for whether or not the modal widget is "open". Maybe "OpenWith" or "SetOpenControl" or something. @essial what do you think?

EDIT: it doesn't need to be addressed in this PR, i think this is fine for now.

**Review by gravestench — APPROVED** [review]
Really good work.

File: hswindow/hsdialog/hsprojectpropertiesdialog/projectpropertiesdialog.go (hunk 2 above)
Comment (ianling) [reply hswindow/hsdialog/hsprojectpropertiesdialog/projectpropertiesdialog.go:81]: The only argument I can think of in favor if keeping it like that is that it's some kind of weird immediate mode factory constructor setup thing, but it doesn't fit with standard golang names and I can see the confusion. I think something like `SetIsOpen` or `SetOpenControl` would be more clear, as well as changing the rest of those names, like `SetSize` and `SetLayout`

**Pull request merged**